### ISSUE TITLE
snowflake_role_grant inconsistent with snowflake_grant_privileges_to_account_role when using escaped characters

### ISSUE THREAD

**Issue by wpl-sf-nbirch** [Snowflake-Labs/terraform-provider-snowflake#2548]
### Terraform CLI and Provider Versions

Terraform v1.7.0
on windows_amd64
+ provider registry.terraform.io/hashicorp/aws v5.37.0
+ provider registry.terraform.io/snowflake-labs/snowflake v0.86.0

### Terraform Configuration

```terraform
locals {
  support_role = "\"USER-EDP.L3.SUPPORT\""
}

resource "snowflake_role" "support_role" {
  provider = snowflake.aad_provisioner
  name    = local.support_role
}

resource "snowflake_role_grants" "support_role_grants" {
  role_name = snowflake_role.support_role.name

  roles = [
    "SNOWFLAKE_ADMIN",
  ]
}
```

### Expected Behavior

This should assign role SNOWFLAKE_ADMIN to role USER-EDP.L3.SUPPORT

### Actual Behavior

Error: '"' can not start an identifier

with module.base_platform.snowflake_role_grants.support_role_grants
on standard_roles.tf line 1506, in resource "snowflake_role_grants" "support_role_grants":

  role_name = snowflake_role.support_role.name

### Steps to Reproduce

1. `terraform plan`
2. `terraform apply`

### How much impact is this issue causing?

High

### Additional Information

Please note that the above configuration snippet has redacted values, and is not in the same format as the actual deployment, which is using modules.

**Comment by sfc-gh-asawicki** [comment]
Hey @wpl-sf-nbirch. I will take a look at it on Monday.

**Comment by sfc-gh-jcieslak** [comment]
Hey @wpl-sf-nbirch. We have released the fix as part of [v0.87.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.87.0) release. Please follow the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/v0.87.0/MIGRATION_GUIDE.md#v0860--v0870) during the update. Please confirm that the issue is resolved in the newest version. Thanks!
